src/emulator/gui/src/shader_editor_dialog.cpp

```diff
@@ -0,0 +1,165 @@
+// Vita3K emulator project
+// Copyright (C) 2018 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#include <gui/functions.h>
+
+#include "private.h"
+#include <crypto/hash.h>
+#include <gxm/types.h>
+#include <host/state.h>
+#include <renderer/state.h>
+#include <shader/spirv_recompiler.h>
+
+#include <imgui_memory_editor.h>
+#include <spdlog/fmt/fmt.h>
+
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include <array>
+#include <cstdio>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+
+#define EXTERNAL_DISASM_ENABLE 0
+#define EXTERNAL_DISASM_PATH ""
+#define EXTERNAL_DISASM_ARGS ""
+
+namespace util {
+
+template <size_t MaxOutSize>
+std::string exec(std::string cmd) {
```

Comment:
```suggestion
std::string exec(const std::string &cmd) {
```

I want to put this in utils. Don't want cross-platform code hanging around in gui.

Reply:
You can still put this here, im just making suggestion

Reply:
yea fair enough was just being lazy tbh